dev/com.ibm.ws.jsf.2.2_fat/fat/src/com/ibm/ws/jsf22/fat/tests/JSF22MiscellaneousTests.java

```diff
@@ -188,46 +204,33 @@ public void testAPI() throws Exception {
-
-            // Make sure the page initially renders correctly
-            if (page == null) {
-                Assert.fail("test ResetValues page was not available.");
-            }
+            ExtendedWebDriver driver = new CustomDriver(new RemoteWebDriver(chrome.getSeleniumAddress(), new ChromeOptions().setAcceptInsecureCerts(true)));
 
-            assertTrue(page.asText().contains("JSF 2.2 Test ResetValues"));
+            String url = JSFUtils.createSeleniumURLString(jsf22MiscellaneousServer, APP_NAME_MISCELLANEOUS, "testResetValues.xhtml");
+            WebPage page = new WebPage(driver);
+            Log.info(c, name.getMethodName(), "Navigating to: /" + APP_NAME_MISCELLANEOUS + "/testResetValues.jsf");
+            page.get(url);
+            page.waitForPageToLoad();
+            assertTrue(page.isInPageText("JSF 2.2 Test ResetValues"));
 
-            // Fill in the FirstName, but not the LastName.   Click "Save", which should
-            //   respond w/ an error.    The first name and last name are both REQUIRED.
-            //   Click "Reset" and this should reset the first name.    Prior to JSF2.2, the
-            //   first name still held the text.
-            HtmlTextInput firstName = (HtmlTextInput) page.getElementById("form1:firstName");
-            firstName.type("John");
+            WebElement firstNameInupt = page.findElement(By.id("form1:firstName"));
+            firstNameInupt.sendKeys("John");
 
-            //  Save the form which should cause an error to be reported.
-            HtmlElement button = (HtmlElement) page.getElementById("form1:save");
-            page = button.click();
+            page.findElement(By.id("form1:save")).click();
+            page.waitReqJs();
+            assertTrue(page.findElement(By.id("form1:firstName")).getAttribute("value").contains("John"));
 
-            if (!page.asText().contains("John")) {
-                Assert.fail("Invalid response from server.  First Name is NOT set to <John>  = " + page.asXml());
-            }
+            WebElement reset = page.findElement(By.id("form1:reset"));
+            Log.info(c, name.getMethodName(), "Clicking Reset...");
+            reset.click();
+            page.waitReqJs();
 
-            //  Click "reset", which should remove "John" from the form.
-            button = (HtmlElement) page.getElementById("form1:reset");
-            page = button.click();
+            String input = page.findElement(By.id("form1:firstName")).getAttribute("value");
+            Log.info(c, name.getMethodName(), "form1:firstName value (expecting empty string) = " + input);
 
-            // Look for the correct results.   The "John" text should not exist in the page.
-            if (page.asText().contains("John")) {
-                Assert.fail("Invalid response from server.  First Name is still set to <John>  = " + page.asXml());
-            }
-        }
+            assertFalse(input.contains("John"));
```

Comment:
This test was problematic, as the last assert saw John in the page even though I clicked the reset button. I refactored it a bit to the check the `form1:firstName` element value, and it works more consistently now.

Reply:
What does the failing assert look like now we don't have the original failure message? I just want to ensure it is easy to debug a failure in the future. Any idea what was sticking around causing the refactoring to be necessary?

Reply:
The assert is still the same: we are ensuring "John" is not in the page after clicking reset 
`            assertFalse(input.contains("John"));`

Previously, I tried searching the page for "John", but it kept finding it.  Selenium wasn't updating the page properly for some reason. Instead, I updated the test to check the form1:firstName element instead.  That's when it started passing consistently since "John" was not present.